src/lib/structures/expr.ml

```diff
@@ -1438,10 +1436,27 @@ and mk_forall_bis (q : quantified) =
         let q = {q with binders; user_trs = trs; sko_v; main = f } in
         mk_forall_bis q
 
+(* Look for a substitution [sbt] of term variables such that:
+   - the domain of [sbt] is a subset of [binders];
+   - the application of [sbt] on the formula [f] produces a formula [g]
+     such that:
+     -- the set of free term variables of [g] in [binders] is strictly smaller;
+     -- the universal closures of [f] and [g] are equivalent.
```

Comment:
My understanding was that it we are only trying to transform implications of the form `x = e -> P(x)` into `P(e)`; is it actually more general?

Reply:
It's a bit more general because it transforms the formula `∀ x : bool, x  -> Q(x)` into `Q(true)`.

I wrote this paragraph to explain why this transformation is valid, but I can write an exhaustive list of the transformations. I didn't think in terms of implications, which is simpler to understand indeed.

Reply:
Right but if `x` is a boolean, `x` is actually `x = true` (: 

Having the general context where the transformation takes place is useful but I'd say explaining what the function actually does is more valuable.